File: Sort_By_Height/Sort_by_height.js
function solution(array) {
	var heights = []

	array.forEach(num=> {
		if(num!==-1)
			heights.push(num)
	})

	for (let j = 0; j < heights.length - 1; j++) {
		for (let i = 0; i < heights.length - 1 -j; i++) {
			const current = heights[i]
			if(current > heights[i+1]){
				heights[i] = heights[i+1]
				heights[i+1] = current
			}
		}
	}

	for (let i = 0, j = 0; i < array.length; i++) {
		array[i] = array[i] !== -1? heights[j++]: -1
	}
	return array 
}
const tests = [
	[-1, 150, 190, 170, -1, -1, 160, 180],
	[-1],
	[23, 54, -1, 43, 1, -1, -1, 77, -1, -1, -1, 3]
];

for (const test of tests) {
  console.log(` unsorted: ${test} \n sorted: ${solution(test)}\n\n`)
}

File: Sort_By_Height/Sort_by_height.java
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
public class Sort_by_height{
	public static void main (String[] args){
            int [][] tests = {
                {-1, 150, 190, 170, -1, -1, 160, 180},
                {-1},
                {23, 54, -1, 43, 1, -1, -1, 77, -1, -1, -1, 3}
            };
            for(int[] test : tests){
                System.out.format(" Non sorted: %s %n sorted: %s %n%n", Arrays.toString(test), Arrays.toString(solution(test)));
            }
	}

	static int[] solution(int[] a){
            List<Integer> heights = new ArrayList<>(a.length);

            for (int num : a)
                if (num != -1)
                    heights.add(num);

            for (int j = 0; j < heights.size() - 1; j++){
                for (int i = 0; i < heights.size() - 1 - j; i++) {
                    int current = heights.get(i);
                    int next = heights.get(i+1);
                    if ( current > next) {
                        heights.set(i, next);
                        heights.set(i + 1, current);
                    }
                }
            }
            
            for (int i = 0, j = 0; i < a.length; i++)
                a[i] = a[i] == -1 ? -1 : heights.get(j++);

            return a;    
        }
}
